core/embed/sys/linker/stm32u58/kernel.ld

```diff
@@ -4,22 +4,23 @@ ENTRY(reset_handler)
 
 MEMORY {
   FLASH  (rx)  : ORIGIN = KERNEL_START, LENGTH = KERNEL_MAXSIZE
-  SRAM1  (wal) : ORIGIN = MCU_SRAM2, LENGTH = 0K
-  BOOT_ARGS  (wal) : ORIGIN = MCU_SRAM2 - BOOTARGS_SIZE, LENGTH = BOOTARGS_SIZE
-  SRAM2  (wal) : ORIGIN = MCU_SRAM2, LENGTH =  KERNEL_SRAM2_SIZE
-  SRAM2_U  (wal) : ORIGIN = MCU_SRAM2 + KERNEL_SRAM2_SIZE, LENGTH = KERNEL_U_RAM_SIZE
-  SRAM3  (wal) : ORIGIN = MCU_SRAM3 + MCU_SRAM3_SIZE - FRAMEBUFFER_SRAM_SIZE, LENGTH = FRAMEBUFFER_SRAM_SIZE
-  SRAM5  (wal) : ORIGIN = MCU_SRAM5, LENGTH =  0K  /* SRAM5 is not available */
-  SRAM6  (wal) : ORIGIN = MCU_SRAM6, LENGTH =  0K  /* SRAM6 is not available */
-  SRAM4  (wal) : ORIGIN = MCU_SRAM4, LENGTH =  MCU_SRAM4_SIZE
+
+  MAIN_RAM (wal) : ORIGIN = MAIN_RAM_START, LENGTH = MAIN_RAM_SIZE
+  BOOT_ARGS (wal) : ORIGIN = BOOTARGS_START, LENGTH = BOOTARGS_SIZE
+  KERNEL_URAM (wal) : ORIGIN = KERNEL_URAM_START, LENGTH = KERNEL_URAM_SIZE
+  FB1_RAM (wal) : ORIGIN = FB1_RAM_START, LENGTH = FB1_RAM_SIZE
+  FB2_RAM (wal) : ORIGIN = FB2_RAM_START, LENGTH = FB2_RAM_SIZE
+
+  COREAPP_RAM (wal) : ORIGIN = AUX_RAM_START, LENGTH = AUX_RAM_SIZE
```

Comment:
What about removing `COREAPP_RAM` and `COREAPP_RAM2`, since we don't use these two memory regions (we don't place anything into) in the kernel?

Reply:
i agree that this is more in line with the new approach, so refactored further here https://github.com/trezor/trezor-firmware/commit/c4d3ed73f4f5698cd31b8ae8c0161dd519190a4a 

we used constants from linker script for initializing the applet, so this required some changes, also in F4 models which i originally didn't want to touch.